Resolved net::ERR_CONNECTION_REFUSED error in Dockerized environment by:

Shaimaa01 opened the pull request:
- Configuring Vite proxy in vite.config.ts to forward '/api' requests to 'http://backend:3000', enabling frontend-backend communication within Docker network.
- Updated axiosConfig.ts baseURL from 'http://localhost:3000/api/v1' to '/api/v1' to use relative URLs, leveraging the Vite proxy for correct API routing.

jackli921 committed:
WIP - add zustand

affan880 committed:
Refactor authentication flow and API integration

- Introduced a new API utility for handling generic API calls.
- Updated authentication store to manage login, registration, and user details.
- Enhanced routing logic in App component to differentiate between public and protected routes.
- Added user details form validation schema.
- Updated Login and Register pages to utilize the new authentication store and handle navigation based on user authentication status.
- Adjusted import paths for consistency.
- Conditionally showing user details form.

affan880 committed:
Fix import path for HomePage component in App.tsx

affan880 committed:
Update import paths for consistency in Auth pages and enhance error handling in LoginPage

Shaimaa01 committed:
Resolved net::ERR_CONNECTION_REFUSED error in Dockerized environment by:
- Configuring Vite proxy in vite.config.ts to forward '/api' requests to 'http://backend:3000', enabling frontend-backend communication within Docker network.
- Updated axiosConfig.ts baseURL from 'http://localhost:3000/api/v1' to '/api/v1' to use relative URLs, leveraging the Vite proxy for correct API routing.

jackli921 reviewed (commented):
I'm able to test Affan's changes when running client and server manually without docker compose, but I got some import errors when trying to spin up the services using docker compose. Same errors when running this `feat/side-nav-tabs` locally.

![image](https://github.com/user-attachments/assets/db7d7783-f680-4c39-9d90-ff1e8a2a3f37)

Shaimaa01 committed:
feat/dashboard: Updated home page, added dashboard components, and created style guide
- Updated home page to serve as main dashboard.
- Added sidebar for desktop navigation.
- Added mobile nav for bottom navigation.
- Created tabs with empty dashboard, transactions, budgets, and savings components.
- Added styleGuide.css in styles folder with design colors .

Shaimaa01 committed:
feat(budget): Create AddBudgetButton
- Added button with dialog form to create budgets
- Integrated budget store for data management
- Used constants and types for form options and validation

Shaimaa01 committed:
.

Shaimaa01 committed:
Fixed edit form in BudgetDialog.tsx to pre-fill category field by mapping capitalized budget.category to lowercase CATEGORY_OPTIONS value.

Shaimaa01 committed:
Added logic to BudgetFormFields.tsx to prevent duplicate category and theme selections by disabling and reducing opacity of used options in dropdowns

affan880 commented:
I see lots of conflicts.

I'm getting network error when I try to sign up or login in your branch.

what's the reason behind forwarding '/api' requests to 'http://backend:3000/',

Both the Login and Sign Up are working in Development branch though

Shaimaa01 committed:
Added spending input

Shaimaa01 committed:
deleted unsedfile

Shaimaa01 committed:
Merge development into feat/side-nav-tabs

Shaimaa01 committed:
Calculated totalMaximum, totalSpending, and totalRemaining in budget store.
Added pieChart and spendingSummary components.

jackli921 commented:
There are 14 commits in this one branch called /side-nav, is this all the changes related to the /side-nav?
If not, I think we should split up all the individual changes as separate PRs that just include the changes of that PR

kayoMichael commented on package.json:
```diff
@@ -1,46 +0,0 @@
-{
-  "name": "monorepo-root",
```
We should not delete Package.json!!! Its still used for bun.

Shaimaa01 committed:
feat(nav): Add logout button to Sidebar and MobileNav with navigation to landing page

Shaimaa01 committed:
chore: merge development and resolve conflict

Shaimaa01 commented:
@jackli921  the branch not have only the sidenav but also the budget page almost all features but the problem of log in still the same network error I made a {/* <Route path="/" element={<HomePage />} /> */} just temporary to be able to see the home page and develop it

affan880 committed:
chore: update bun.lock

kayoMichael committed:
Refactor Auth flow and change eslint config

kayoMichael committed:
Resolve tests

kayoMichael committed:
Refactor Workflow

kayoMichael committed:
turn off samesite for preview
